monai/data/png_writer.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from skimage import io, transform
+
+
+def write_png(data,
+              file_name,
+              output_shape=None,
+              interp_order=3,
+              mode='constant',
+              cval=0,
+              scale_factor=255):
+    """
+    Write numpy data into png files to disk.  
+    Spatially It supports HW for 2D.(H,W) or (H,W,3) or (H,W,4)
+
+    Args:
+        data (numpy.ndarray): input data to write to file.
+        file_name (string): expected file name that saved on disk.
+        output_shape (None or tuple of ints): output image shape.
+        interp_order (int): the order of the spline interpolation, default is 3.
+            The order has to be in the range 0 - 5.
+            this option is used when `output_shape != None `.
+        mode (`reflect|constant|nearest|mirror|wrap`):
+            The mode parameter determines how the input array is extended beyond its boundaries.
+            this option is used when `output_shape != None`.
+        cval (scalar): Value to fill past edges of input if mode is "constant". Default is 0.0.
+            this option is used when `output_shape != None`.
+    """
+    assert isinstance(data, np.ndarray), 'input data must be numpy array.'
+
+    if scale_factor > 0:
+        max_val = np.max(data)
+        min_val = np.min(data)
+        data = (data - min_val) / (max_val - min_val)
+
+    if output_shape is not None:
+        assert isinstance(output_shape, (list, tuple)) and len(output_shape) == 2, \
+            'output_shape must be a list of 2 values (H, W).'
+
+        if len(data.shape) == 3:
+            output_shape += (data.shape[2],)
+
+        data = transform.resize(data, output_shape, order=interp_order, mode=mode, cval=cval)
+
+    data = scale_factor * data 
```

Comment:
`data = scale_factor * data` should in the `if scale_factor > 0:`?

Reply:
I think we should just have 
`if scale_factor > 0:
    data = scale_factor * data`  in line 55. That will do the trick. transform resize operates on 0-1 range.

Reply:
Sounds good, I think we don't need to scale only on the data in [0 - 1] range, this is users' responsibility to use scale on proper data. So I suggest changing the default value of `scale_factor` from 255 to 0.
Thanks.

Reply:
Do we have to do 0-1 normalisation ? We are currently doing that using min and max. If we remove this, then it becomes more like it's user responsibility to provide the data as integer or bool format.

Reply:
OK, you are right. I reviewed this problem again today, I think maybe we can make an easier version first:
1. if want to scale, assert that `min` must >= 0 and `max` must <= 1, don't do normalization with min and max anymore.
2. scale factor will be only several fixed values, like: None, 255, 65525, etc.(@wyli may share more comments).
3. convert to uint8 or uint16 according to the scale factor.

What do you think?
Let's switch to @wyli 's comment for further discussion, I think these `resize` and `scale` are connected issues.

Thanks.